configure.ac

```diff
@@ -74,10 +111,39 @@ AM_CONDITIONAL([SYSTEMD_CONDITION], [test "$SYSTEMD_STR" = yes])
 # target directory for ledmon service file
 AC_SUBST([SYSTEMD_PATH], "$(pkg-config systemd --variable=systemdsystemunitdir)")
 
+
+AC_ARG_WITH([test],
+    [AS_HELP_STRING([--without-test],
+                    [disable all test case])],
+    [], [with_test=yes])
+
+AM_CONDITIONAL([WITH_TEST], [test "x$with_test" = "xyes"])
+
+if test "x${with_test}" = "xyes"; then
+    PKG_CHECK_MODULES([LIBCHECK], [check >= 0.9.8 ])
+fi
+
+# Add configure option to build without library
+AC_ARG_ENABLE([library],
+    [AS_HELP_STRING([--disable-library],
+        [disable building ledmon library])],
+    [with_library=${enableval}],
+    [with_library=yes])
+
+AM_CONDITIONAL([WITH_LIBRARY], [test "x$with_library" = "xyes"])
```

Comment:
> Ideally, we should provide `=yes/no` values in `./configure` line to keep maximum consistency:
> 
> ```
> # ./configure --enable-systemd=yes --enable-library=yes --with-test=yes
> # ./configure --enable-systemd=no --enable-library=no --with-test=no
> ```

This exact syntax works with this PR as it is ...


However, I've updated the PR so that each one of these options is identical in behavior with them defaulting to being disabled.   Some examples with the latest updates...

```
$ ./configure --help
...
  --enable-systemd        install ledmon systemd service
  --enable-test           enable test cases, run using 'make check'
  --enable-library        enable building ledmon library
```

Examples
```
$ ./configure
...
configure parameters:    --enable-systemd=no --enable-library=no --enable-test=no

$ ./configure --enable-systemd --enable-library --enable-test
...
configure parameters:    --enable-systemd=yes --enable-library=yes --enable-test=yes

$ ./configure --enable-systemd=no --enable-library=no --enable-test=no
...
configure parameters:    --enable-systemd=no --enable-library=no --enable-test=no

$ ./configure --enable-systemd=yes --enable-library=yes --enable-test=yes
...
configure parameters:    --enable-systemd=yes --enable-library=yes --enable-test=yes

```